packages/core/src/agent/Agent.ts

```diff
@@ -304,4 +298,57 @@ export class Agent {
     }
     return connection
   }
+
+  private registerDependencies(dependencyManager: DependencyManager) {
+    dependencyManager.registerInstance(AgentConfig, this.agentConfig)
+
+    dependencyManager.registerSingleton(EventEmitter)
+    dependencyManager.registerSingleton(MessageSender)
+    dependencyManager.registerSingleton(MessageReceiver)
+    dependencyManager.registerSingleton(TransportService)
+    dependencyManager.registerSingleton(Dispatcher)
+    dependencyManager.registerSingleton(EnvelopeService)
+
+    // Register possibly already defined services
```

Comment:
Did you check if there is a way to avoid this? I think it can get quite long, and something that can be quickly forgotten to do when adding a new dependency.

Reply:
No there's no way around this, if you're adding a service you should register it. That's the only way we can register it with the correct lifecycle scope.